src/peft/peft_model.py

```diff
@@ -271,15 +271,18 @@ def _prepare_model_for_gradient_checkpointing(self, model: PreTrainedModel):
         r"""
         Prepares the model for gradient checkpointing if necessary
         """
-        if not (getattr(model, "is_loaded_in_8bit", False) or getattr(model, "is_loaded_in_4bit", False)):
-            if hasattr(model, "enable_input_require_grads"):
-                model.enable_input_require_grads()
-            else:
+        try:
+            if not (getattr(model, "is_loaded_in_8bit", False) or getattr(model, "is_loaded_in_4bit", False)):
+                if hasattr(model, "enable_input_require_grads"):
+                    model.enable_input_require_grads()
+                else:
 
-                def make_inputs_require_grad(module, input, output):
-                    output.requires_grad_(True)
+                    def make_inputs_require_grad(module, input, output):
+                        output.requires_grad_(True)
 
-                model.get_input_embeddings().register_forward_hook(make_inputs_require_grad)
+                    model.get_input_embeddings().register_forward_hook(make_inputs_require_grad)
+        except:
+            pass
```

Comment:
I don't think this is needed as the `getattr(model, "is_loaded_in_8bit", False)` already protects from the case where the model doesn't have that attribute no? I would say maybe let's avoid adding bare try / except blocks in the code base.

Reply:
for some reason this was leading to issues when using Auto1111 webui with PEFT which uses CompVis implementation of SD.

Reply:
Now I remember,  Compvis SD is when loaded in float16, will go to the `else` block and it won't have attribute `get_input_embeddings ` leading to error. @younesbelkada, WDYT?

Reply:
but wanted to bring this to your notice when working with non transformers models

Reply:
Note that I made exactly that change in my PR for custom models:

https://github.com/huggingface/peft/pull/676/files#diff-95787fcc5356568929e383449ade2c1bac5da43deccab17318652b62ed171ae7R276

I agree this is better than bare `except` (and _if_ we need to use bare `except`, it should always be `except Exception` in order not to catch `KeyboardInterrupt`).

Reply:
Cool, will remove this fix as you are already doing it. Also, thanks for fixing the config part too as I was setting a dummy config manually for my usecase.